controllers/tektontasks_controller.go

```diff
@@ -175,17 +176,37 @@ func (r *tektonTasksReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+type reconcileResult struct {
+	result []common.ReconcileResult
+	err    error
+}
+
 func (r *tektonTasksReconciler) reconcileOperands(tektonRequest *common.Request) ([]common.ReconcileResult, error) {
 	// Reconcile all operands
 	allReconcileResults := make([]common.ReconcileResult, 0, len(r.operands))
-	for _, operand := range r.operands {
+	results := make([]reconcileResult, len(r.operands))
+
+	wg := sync.WaitGroup{}
+	for i, operand := range r.operands {
 		tektonRequest.Logger.V(1).Info(fmt.Sprintf("Reconciling operand: %s", operand.Name()))
-		reconcileResults, err := operand.Reconcile(tektonRequest)
-		if err != nil {
-			tektonRequest.Logger.Info(fmt.Sprintf("Operand reconciliation failed: %s", err.Error()))
-			return nil, err
+
+		wg.Add(1)
+		go func(operand operands.Operand, index int) {
+			defer wg.Done()
+			reconcileResults, err := operand.Reconcile(tektonRequest)
+			results[index].err = err
+			results[index].result = append(results[index].result, reconcileResults...)
```

Comment:
Why not directly assign it?

`results[index].result, results[index].err = operand.Reconcile(tektonRequest)`